### DISCUSSION TITLE
[V4] Prose

### DISCUSSION THREAD

**Discussion by NicolasCassunde** [tailwindlabs/tailwindcss#14120] [Help] [answered]
How can i use PROSE typography in tailwind v4?

**Comment by florence0239** [comment]
> How can i use PROSE typography in tailwind v4? [hpinstantink](https://www.hpinstantink.com.co)

Hello,

I'm also looking to use PROSE typography with Tailwind v4. Can anyone provide guidance on how to integrate this into my project? Thanks!


Best regards,
florence023

**Comment by apz-gh** [comment]
I've been importing the stylesheet shared by Adam Wathan in [this tweet](https://x.com/adamwathan/status/1757850347777687870):

**Gist:** [adamwathan/41dcab602afd07ac8ba243eb186c324a](https://gist.github.com/adamwathan/41dcab602afd07ac8ba243eb186c324a)

**Comment by cbmongithub** [comment]
# Customizing Tailwindcss V4 Alpha
## MDX, Prose, and Highlight Elements 

### Via `tailwindcss/cli@next`

Use the empty component layer for component related styles. Create a `styles` folder, along with nested base.css, theme.css, components.css, utilities.css, and finally, index.css files. Copy styles from tailwind repo. Init the cli if you haven't already. 

Put your typography styles in `components.css` or `base.css` at the bottom, and the following in `index.css` :

```css
@layer theme, base, components, utilities;

@import "./theme.css" layer(theme);
@import "./base.css" layer(base);
@import "./components.css" layer(components);
@import "./utilities.css" layer(utilities);

```

Configure `@tailwindcss/cli@next` with a script or run the init function in the cli. I personally use pnpm scripts to build, watch, optimize and minify. I use nextjs and output to the default globals.css file. I also use concurrently and run the tw scripts in production mode, per source of `tailwindcss/postcss` [plugin](https://github.com/tailwindlabs/tailwindcss/blob/next/packages/%40tailwindcss-postcss/src/index.ts#L42)

```json
    "tw:build": "NODE_ENV=production && tailwindcss -i ./styles/index.css -o app/globals.css --optimize -m",
    "tw:watch": "NODE_ENV=production && tailwindcss -i ./styles/index.css -o app/globals.css --optimize -m -w",
    "develop": "concurrently --names \"CSS,DEV\" -c \"bgBlue.bold,bgMagenta.bold\" -k \"pnpm tw:watch\" \"pnpm dev\""
```

Finally, run the script: 

```bash
pnpm/npm run develop
```

### Via `tailwindcss@next` (Customized defaults, fastest way imo)

```bash
pnpm/npm install tailwindcss@next
```
Follow the v4 tutorial in the tailwind blog and configure it up until the point where you get to globals.css file or whatever index file it is. You want to do what we did with the index file above, in the globals.css file (nextjs), aside from changing source. It's been a few months, but it should go something like this:

```css
@layer theme, base, components, utilities;

@import "tailwindcss/theme" layer(theme);
@import "tailwindcss/preflight" layer(base);
@import "tailwindcss/components" layer(components);
@import "tailwindcss/utilities" layer(utilities); 

@layer components {
... 
...
}
```

Or whichever layer you want to append styles to. Just remember to pay attention to ***cascade order*** and don't forget to configure your postcss config file. Make sure plugin is set to `"@tailwindcss/postcss": {}` or `"@tailwindcss/postcss": {base: 'your_base_file', optimize: {minify:true}}` 

I believe that's it. Unless things have changed. Just refer to source and your answers are there, or you can reply and I will answer as soon as I am able. I am currently experimenting with all sorts of configs so if anyone has something better, please share. 

### References: 

- Link to [tailwindcss/next branch](https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/index.css)
- Link to [tailwindcss blog](https://tailwindcss.com/blog/tailwindcss-v4-alpha)

I hope this helps you and other fellow devs!

`@_coderchris` on 𝕏 if you'd like 👍

**Comment by aaronadamsCA** [comment]
You can write your own basic prose utility until somebody, hopefully Tailwind, ships one for v4!

Here's something to get you started: https://play.tailwindcss.com/4d11rHJifI?file=css

![image](https://github.com/user-attachments/assets/bd7a8002-1492-47a6-a721-704a870b6e4e)

This has the enormous advantage of being zero-config. If you're on Tailwind v4 latest, you can write a `@utility` in your `tailwind.css` and you're done.

**Comment by fcrozatier** [comment, marked as answer]
You can add the typography plugin like this in your css

```css
@import "tailwindcss";
@plugin '@tailwindcss/typography';
```

**Reply by cbmongithub** [reply]
Yeah they added that after :P seen that. Obv the best solution. It's all coming together and I am so stoked for it.

**Comment by rozsazoltan** [comment]
References:
* [How to configure @tailwindcss/typography plugin from TailwindCSS v4](https://stackoverflow.com/a/79626930/15167500)
* [Typography upgrade to TailwindCSS v4 - @ErwannRousseau's code snippet](https://github.com/tailwindlabs/tailwindcss-typography/issues/372#issuecomment-2616848354)
